oaknational/Oak-Web-Application
# Feat/lesq 1054/hubspot checkbox conditional [LESQ-1054]

## Description

Music year: {owa_music_year}

- Add an api route to get a user's subscription status
- Make the hubspot sign up checkbox conditional on subscription status

## How to test

1. Go to https://deploy-preview-2756--oak-web-application.netlify.thenational.academy/onboarding **while logged in**
2. You should not see the sign up checkbox
3. Go to school selection
4. You should see an unchecked checkbox on the page (if you haven't already set up a subscription)
5. Go to the contact for your email in hubspot ([example](https://app-eu1.hubspot.com/contacts/24947711/record/0-1/33730277100)) and on the left under 'communication subscriptions` update the School Support subscription to subscribed 
6. Go back to the school selection page and refresh
7. You should not see the checkbox

## Screenshots

How it used to look (d
<img width="448" alt="Screenshot 2024-09-05 at 16 04 41" src="https://github.com/user-attachments/assets/40800b52-463c-4ad7-b115-35bf1cf49e19">
elete if n/a):


How it should now look:
Not subscribed:
<img width="569" alt="Screenshot 2024-09-05 at 15 32 07" src="https://github.com/user-attachments/assets/c3e58c03-68e0-469b-a690-06cb5ad28e86">
Already subscribed:
<img width="509" alt="Screenshot 2024-09-05 at 16 03 45" src="https://github.com/user-attachments/assets/a3bd9407-df93-4b3a-9de7-74c9ec46f007">

## Checklist
- [ ]  When user has already opted in to hubspot marketing (via their email) they will not see a checkbox during onboarding
- [ ]  When a user has no record of signing up in hubspot they will see an unchecked check box during onboarding
